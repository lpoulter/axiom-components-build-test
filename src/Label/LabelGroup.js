import PropTypes from 'prop-types';
import React, { Component } from 'react';
import InlineGroup from '../InlineGroup/InlineGroup';

export default class LabelGroup extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <InlineGroup space="x4" { ...rest } className="ax-label-group">
        { children }
      </InlineGroup>
    );
  }
}
