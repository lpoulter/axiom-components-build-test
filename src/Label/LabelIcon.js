import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Icon from '../Icon/Icon';
import Link from '../Typography/Link';
import './LabelIcon.css';

export const LabelIconRef = 'LabelIcon';

export default class LabelIcon extends Component {
  static propTypes = {
    /** SKIP */
    color: PropTypes.string,
    /** SKIP */
    isEnd: PropTypes.bool,
    /** SKIP */
    isStart: PropTypes.bool,
    /** Name of the icon. See <Icon>. */
    name: PropTypes.string.isRequired,
  };

  static typeRef = LabelIconRef;

  render() {
    const { color, isEnd, isStart, name, ...rest } = this.props;
    const classes = classnames('ax-label__icon', {
      'ax-label__icon--start': isStart,
      'ax-label__icon--end': isEnd,
    });

    const linkStyle = {
      white: 'subtle',
      success: 'night',
      error: 'night',
    }[color];

    return (
      <span className={ classes }>
        <Link { ...rest } style={ linkStyle }>
          <Icon name={ name } size="1rem" />
        </Link>
      </span>
    );
  }
}
