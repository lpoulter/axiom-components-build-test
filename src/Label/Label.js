import PropTypes from 'prop-types';
import React, { Children, Component, cloneElement } from 'react';
import classnames from 'classnames';
import { isComponent } from '@brandwatch/axiom-utils';
import Base from '../Base/Base';
import { LabelIconRef } from './LabelIcon';
import './Label.css';

export default class Label extends Component {
  static propTypes = {
    /** Content inserted into the Label */
    children: PropTypes.node,
    /** Color of the label */
    color: PropTypes.oneOf(['white', 'success', 'error']),
    /**
     * Controls the full width appearance of the badge either all of the time,
     * with a value of `true` otherwise at one of the breakpoints specified.
     */
    full: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /** Size of standard shape */
    size: PropTypes.oneOf(['small', 'medium']),
  };

  static defaultProps = {
    color: 'white',
    size: 'medium',
  };

  render() {
    const {
      children,
      color,
      full,
      size,
      ...rest
    } = this.props;

    const classes = classnames('ax-label', {
      [`ax-label--${color}`]: color,
      [`ax-label--${size}`]: size,
      'ax-label--full': full === true,
      [`ax-label--full--${full}`]: typeof full === 'string',
    });

    const mappedChildren = Children.toArray(children).map((child, index, array) =>
      !isComponent(child, LabelIconRef) ? child : cloneElement(child, {
        isEnd: index === array.length - 1,
        isStart: index === 0,
        color,
        size,
      })
    );

    return (
      <Base { ...rest } Component="span" className={ classes }>
        { mappedChildren }
      </Base>
    );
  }
}
