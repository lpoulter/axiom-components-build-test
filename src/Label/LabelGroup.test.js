import React from 'react';
import renderer from 'react-test-renderer';
import Label from './Label';
import LabelGroup from './LabelGroup';

const getComponent = (props = {}) =>
  renderer.create(
    <LabelGroup { ...props }>
      <Label>Lorem ipsum</Label>
    </LabelGroup>
  );

describe('LabelGroup', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
