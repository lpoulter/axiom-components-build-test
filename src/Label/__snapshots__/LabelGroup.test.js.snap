// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelGroup renders with defaultProps 1`] = `
<div
  className="ax-inline-group ax-label-group ax-space--x4"
>
  <div
    className="ax-inline-group__spacer"
  >
    <span
      className="ax-label ax-label--white ax-label--medium"
    >
      Lorem ipsum
    </span>
  </div>
</div>
`;
